### PR TITLE
Add slcli vs create by router data.

### PR DESCRIPTION
**Pull request by FernandoOjeda**
Add slcli vs create by router data https://github.com/softlayer/softlayer-python/issues/1412.

### CODE REVIEW

File: SoftLayer/managers/vs.py [hunk 1]
```diff
@@ -533,10 +533,7 @@ def _generate_create_dict(
         if datacenter:
             data["datacenter"] = {"name": datacenter}
 
-        if private_vlan or public_vlan or private_subnet or public_subnet:
-            network_components = self._create_network_components(public_vlan, private_vlan,
-                                                                 private_subnet, public_subnet)
-            data.update(network_components)
+        self.get_network_components(data, kwargs, private_subnet, private_vlan, public_subnet, public_vlan)
```
Comment (allmightyspiff) [inline SoftLayer/managers/vs.py:536]: I don't like function that modify the data that gets passed into them. I'd prefer this function `get_network_componets` return `data` instead.

So this line would be:
```
network_components = self.get_network_components(kwargs, private_subnet, private_vlan, public_subnet, public_vlan)
data.update(network_components)
```
as that reads better to me.

File: SoftLayer/managers/vs.py [hunk 2]
```diff
@@ -579,9 +576,34 @@ def _generate_create_dict(
 
         return data
 
+    def get_network_components(self, data, kwargs, private_subnet, private_vlan, public_subnet, public_vlan):
+        """Get the network components structure.
+
+        :param data: Array variable to add the network structure.
```
Comment (allmightyspiff) [inline SoftLayer/managers/vs.py:582]: Have this be the `Return` section

File: SoftLayer/managers/vs.py [hunk 3]
```diff
@@ -579,9 +576,34 @@ def _generate_create_dict(
...
+        :param kwargs: Vs item list.
+        :param int private_subnet: Private subnet id.
+        :param int private_vlan: Private vlan id.
+        :param int public_subnet: Public subnet id.
+        :param int public_vlan: Public vlan id.
+        """
+        if kwargs.get('private_router') or kwargs.get('public_router'):
```
Comment (allmightyspiff) [inline SoftLayer/managers/vs.py:589]: For readability, change

```python
if kwargs.get('private_router') or kwargs.get('public_router'):
    if private_vlan or public_vlan or private_subnet or public_subnet:
```
To:

```python
# Throw exception for router option AND vlan/subnet option
if any([kwargs.get('private_router'), kwargs.get('public_router')] and any([private_vlan, public_vlan, private_subnet, public_subnet]):

```

File: SoftLayer/managers/vs.py [hunk 4]
```diff
@@ -579,9 +576,34 @@ def _generate_create_dict(
 
         return data
 
+    def get_network_components(self, data, kwargs, private_subnet, private_vlan, public_subnet, public_vlan):
```
Comment (allmightyspiff) [inline SoftLayer/managers/vs.py:579]: I think you could combine this function with `_create_network_components`,  `kwargs.get()` should rerturn `None` if the option wasn't specified. So really you just need to move the Exception where your checking for vlan/subnets and router to `create_network_components` and save having to make another function.

**Review by allmightyspiff — CHANGES_REQUESTED** [review]
Looks good, but I would recommend combining `get_network_components` and `create_network_components` into one function. Basically just add your check for the exception to `create_network_components` and I think it wil work as you expect.

File: SoftLayer/managers/vs.py (hunk 1 above)
Comment (FernandoOjeda) [reply SoftLayer/managers/vs.py:536]: I did it the same lol, but many unit tests failed. I am going to check it again.

**Review by ATGE — APPROVED** [review]
LGTM

File: SoftLayer/managers/vs.py (hunk 1 above)
Comment (FernandoOjeda) [reply SoftLayer/managers/vs.py]: Done.

File: SoftLayer/managers/vs.py (hunk 3 above)
Comment (FernandoOjeda) [reply SoftLayer/managers/vs.py]: Done.

File: SoftLayer/managers/vs.py (hunk 4 above)
Comment (FernandoOjeda) [reply SoftLayer/managers/vs.py]: Done.

File: SoftLayer/managers/vs.py (hunk 2 above)
Comment (FernandoOjeda) [reply SoftLayer/managers/vs.py]: Done.

**Review by allmightyspiff — APPROVED** [review]
Looks good, thanks

**Review by caberos — APPROVED** [review]
Good job

**Pull request merged**